spock-specs/src/test/groovy/org/spockframework/docs/extension/RetryDocSpec.groovy

```diff
@@ -0,0 +1,110 @@
+package org.spockframework.docs.extension
+
+import groovy.sql.Sql
+import spock.lang.Retry
+import spock.lang.Shared
+import spock.lang.Specification
+
+abstract
+// tag::example-common[]
+class FlakyIntegrationSpec extends Specification {
+// end::example-common[]
+  @Shared
+  def sql = Sql.newInstance("jdbc:h2:mem:", "org.h2.Driver")
+
+// tag::example-d[]
+  @Retry(exceptions = IllegalArgumentException, count = 2)
```

Comment:
Should we add a test, that actually alternately throws both these exceptions?

Reply:
That is unfortunately a good idea, though in the test spec, not the doc spec.
Unfortunately, because I think the changes were too naive for the retry extension logic to properly work in that case in its current state. I have to rework that probably when I'm a bit less sleepy.

Reply:
Or you might have a look, as you invented the whole extension and probably know the code better than me.